compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmInstructionGen.java

```diff
@@ -598,36 +603,80 @@ private BType getSmallestBuiltInUnsignedIntSubTypeContainingTypes(BType lhsType,
+                boolean isInterface = callIns.invocationType == INVOKEINTERFACE;
+                int argIndex = 0;
+                String jMethodVMSig = callIns.jMethodVMSig;
+                boolean hasBalEnvParam = jMethodVMSig.startsWith(BAL_ENV_PARAM);
+                if (hasBalEnvParam) {
+                    mv.visitTypeInsn(NEW, BAL_ENV_CLASS);
+                    mv.visitInsn(DUP);
+                    // load the strand
+                    this.mv.visitVarInsn(ALOAD, localVarOffset);
+                    // load the current Module
+                    mv.visitFieldInsn(GETSTATIC, this.moduleInitClass, CURRENT_MODULE_VAR_NAME, GET_MODULE);
+                    mv.visitMethodInsn(INVOKESPECIAL, BAL_ENV_CLASS, JVM_INIT_METHOD,
+                            INIT_BAL_ENV, false);
+                }
+                while (argIndex < callIns.args.size()) {
+                    BIROperand arg = callIns.args.get(argIndex);
+                    this.loadVar(arg.variableDcl);
+                    argIndex += 1;
+                }
+                this.mv.visitMethodInsn(callIns.invocationType, callIns.jClassName, callIns.name, jMethodVMSig,
+                        isInterface);
+                if (callIns.lhsOp != null) {
+                    this.storeToVar(callIns.lhsOp.variableDcl);
+                }
+                break;
+            case LARGE_ARRAY:
+                JLargeArrayInstruction inst = (JLargeArrayInstruction) ins;
+                BType instType = JvmCodeGenUtil.getReferredType(inst.type);
+                if (instType.tag == TypeTags.ARRAY) {
+                    this.mv.visitTypeInsn(NEW, ARRAY_VALUE_IMPL);
+                    this.mv.visitInsn(DUP);
+                    jvmTypeGen.loadType(this.mv, inst.type);
+                    loadListInitialValues(inst);
+                    BType elementType = JvmCodeGenUtil.getReferredType(((BArrayType) instType).eType);
+                    if (elementType.tag == TypeTags.RECORD || (elementType.tag == TypeTags.INTERSECTION &&
+                            ((BIntersectionType) elementType).effectiveType.tag == TypeTags.RECORD)) {
+                        visitNewRecordArray(elementType);
+                    } else {
+                        this.mv.visitMethodInsn(INVOKESPECIAL, ARRAY_VALUE_IMPL, JVM_INIT_METHOD,
+                                INIT_ARRAY, false);
+                    }
+                    this.storeToVar(inst.lhsOp.variableDcl);
+                } else {
+                    this.loadVar(inst.typedescOp.variableDcl);
+                    this.mv.visitVarInsn(ALOAD, localVarOffset);
+                    loadListInitialValues(inst);
+                    this.mv.visitMethodInsn(INVOKEINTERFACE, TYPEDESC_VALUE, INSTANTIATE_FUNCTION, INSTANTIATE, true);
+                    this.storeToVar(inst.lhsOp.variableDcl);
+                }
+                break;
+            default:
+                JLargeMapInstruction mapNewIns = (JLargeMapInstruction) ins;
```

Comment:
Is it always correct to use default block for `JLargeMapInstruction` here? Can we have a separate case for it?

Reply:
Yes, Since we have only 4 kinds of `JInstruction` we can make `JLargeMapInstruction` the default case. If we add a separate case it will reduce the code coverage.